zapcore/json_encoder.go

```diff
@@ -82,6 +82,9 @@ func NewJSONEncoder(cfg EncoderConfig) Encoder {
 }
 
 func newJSONEncoder(cfg EncoderConfig, spaced bool) *jsonEncoder {
+	if cfg.Precision == 0 {
```

Comment:
What if users want precision = 0?

Reply:
I've changed other codes according to comments up there except this one.
Actually, if the user did not set `FloatPrecision`, the value of `FloatPrecision` here is 0, I can not judge if the user wants 0 or the default value.
Do you have some good suggestions?